libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -465,341 +462,246 @@ public ElasticsearchEntitlementChecker(PolicyManager policyManager) {
+        policyChecker.checkOutboundNetworkAccess(callerClass);
     }
 
     @Override
     public void check$java_net_Socket$bind(Class<?> callerClass, Socket that, SocketAddress endpoint) {
-        policyManager.checkOutboundNetworkAccess(callerClass);
+        policyChecker.checkOutboundNetworkAccess(callerClass);
     }
 
     @Override
     public void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint) {
-        policyManager.checkOutboundNetworkAccess(callerClass);
+        policyChecker.checkOutboundNetworkAccess(callerClass);
     }
 
     @Override
     public void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint, int backlog) {
-        policyManager.checkOutboundNetworkAccess(callerClass);
-    }
-
-    @SuppressWarnings("deprecation")
-    private URL extractJarFileUrl(URL jarUrl) {
-        String spec = jarUrl.getFile();
-        int separator = spec.indexOf("!/");
-
-        // URL does not handle nested JAR URLs (it would be a MalformedURLException upon connection)
-        if (separator == -1) {
-            return null;
-        }
-
-        try {
-            return new URL(spec.substring(0, separator));
-        } catch (MalformedURLException e) {
-            return null;
-        }
-    }
-
-    private boolean handleNetworkOrFileUrlCheck(Class<?> callerClass, URL url) {
-        if (isNetworkUrl(url)) {
-            policyManager.checkOutboundNetworkAccess(callerClass);
-            return true;
-        }
-        if (isFileUrl(url)) {
-            checkURLFileRead(callerClass, url);
-            return true;
-        }
-        return false;
-    }
-
-    private void checkJarURLAccess(Class<?> callerClass, JarURLConnection that) {
-        var jarFileUrl = that.getJarFileURL();
-        if (handleNetworkOrFileUrlCheck(callerClass, jarFileUrl)) {
-            return;
-        }
-        policyManager.checkUnsupportedURLProtocolConnection(callerClass, jarFileUrl.getProtocol());
-    }
-
-    private void checkEntitlementForUrl(Class<?> callerClass, URL that) {
-        if (handleNetworkOrFileUrlCheck(callerClass, that)) {
-            return;
-        }
-        if (isJarUrl(that)) {
-            var jarFileUrl = extractJarFileUrl(that);
-            if (jarFileUrl == null || handleNetworkOrFileUrlCheck(callerClass, jarFileUrl) == false) {
-                policyManager.checkUnsupportedURLProtocolConnection(callerClass, "jar with unsupported inner protocol");
-            }
-        } else {
-            policyManager.checkUnsupportedURLProtocolConnection(callerClass, that.getProtocol());
-        }
+        policyChecker.checkOutboundNetworkAccess(callerClass);
     }
 
     @Override
     public void check$java_net_URL$openConnection(Class<?> callerClass, java.net.URL that) {
-        checkEntitlementForUrl(callerClass, that);
+        policyChecker.checkEntitlementForUrl(callerClass, that);
```

Comment:
I don't understand why these were moved - they were private to EntitlementChecker, why widen their visibility?

Reply:
(I prefer the way they are, and leave them out of the interface so it just have the "basic blocks", but I'm OK with moving them and exposing them if you feel strongly about it)